### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by pjhartzell**
**Related Issue(s):**
None.

**Description:**
Initial Item and Collection creation.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).
- [x] Examples are added to `examples/`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,46 @@
...
+
+### Command-line usage
+
+To create a STAC `Item`:
+
+```bash
+$ stac esaworldcover create-item tests/data-files/ESA_WorldCover_10m_2020_v100_N66E177_Map/ESA_WorldCover_10m_2020_v100_N66E177_Map.tif examples
+```
+
+To create a STAC `Collection` from a text file containing a list of Landsat scene XML metadata files:
```
Comment (gadomski) [inline README.md:38]: Landsat?

File: examples/collection.json [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+          "value": 95,
+          "description": "Mangroves",
+          "color-hint": "00CF75"
+        },
+        {
+          "value": 100,
+          "description": "Moss and lichen",
+          "color-hint": "FAE6A0"
+        }
+      ],
```
Comment (gadomski) [inline examples/collection.json:109]: This reads _so much_ better than The Old Way™, 👏🏽

File: examples/collection.json [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      "url": "https://esa-worldcover.org/en"
+    },
+    {
+      "name": "Microsoft",
+      "roles": [
+        "host"
+      ],
+      "url": "https://planetarycomputer.microsoft.com"
+    }
```
Comment (gadomski) [inline examples/collection.json:203]: We don't want Microsoft in the stactools package.

File: tests/test_commands.py [hunk 4]
```diff
@@ -1,61 +1,25 @@
...
+class ItemCommandTest(CliTestCase):
 
-    def create_subcommand_functions(self):
-        return [create_ephemeralcmd_command]
+    def create_subcommand_functions(self) -> List[Callable[[Group], Command]]:
+        return [create_esaworldcover_command]
 
-    def test_create_collection(self):
+    def test_create_item(self) -> None:
+        infile = "tests/data-files/ESA_WorldCover_10m_2020_v100_N66E177_Map/ESA_WorldCover_10m_2020_v100_N66E177_Map.tif"  # noqa
```
Comment (gadomski) [inline tests/test_commands.py:18]: You can use `test_data.get_path` to get this path -- not required, but allows tests to pass when not run from the root project directory. https://github.com/stac-utils/stactools/blob/main/src/stactools/testing/test_data.py#L38-L40

**Review by gadomski — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (pjhartzell) [reply README.md:38]: Copy and paste. Busted.

File: tests/test_commands.py (hunk 4 above)
Comment (pjhartzell) [reply tests/test_commands.py:18]: Yep. Fixed.

File: examples/collection.json (hunk 3 above)
Comment (pjhartzell) [reply examples/collection.json:203]: Fixed.

File: tests/test_commands.py (hunk 4 above)
Comment (gadomski) [reply tests/test_commands.py:18]: Did you push the fix?

File: examples/collection.json (hunk 3 above)
Comment (pjhartzell) [reply examples/collection.json:203]: Will that be included in the ETL template, though?

File: README.md (hunk 1 above)
Comment (pjhartzell) [reply README.md:38]: Fixed.

File: examples/collection.json (hunk 3 above)
Comment (gadomski) [reply examples/collection.json:203]: Yeah, that's part of the extra work done on the ETL side.

File: tests/test_commands.py (hunk 4 above)
Comment (pjhartzell) [reply tests/test_commands.py]: Yep.

File: examples/collection.json (hunk 3 above)
Comment (pjhartzell) [reply examples/collection.json]: OK. Will need to fix that in Landsat.

Comment (gadomski) [reply examples/collection.json]: > OK. Will need to fix that in Landsat.

Roger.

**Pull request merged**